src/main/java/eu/europeana/api/recommend/util/RecommendServiceUtils.java

```diff
@@ -0,0 +1,170 @@
+     * for the given entity uri.
+     *
+     * example : ?query=type:EntityBestItemsSet&f&qf=subject:<entityUri>&pageSize=<pageSize>&profile=standard&wskey=
+     * @param entityUri
+     * @param pageSize
+     * @param wskey
+     * @return
+     */
+    public String setApiSearchQuery(String entityUri, String pageSize, String wskey) {
+        StringBuilder query = new StringBuilder("?query=type:EntityBestItemsSet");
+        query.append("&qf=subject:").append(entityUri);
+        query.append("&pageSize=").append(pageSize);
+        query.append("&profile=standard");
+        query.append("&").append("wskey=").append(wskey);
+        return query.toString();
+    }
+
+    // ---------------------- Temporary functionality below to mock entity recommendations
+    private String generateEntitySearchQuery(String entityFullId, int maxResults, String wskey) {
+        StringBuilder s = new StringBuilder(50)
+                .append("?query=text:\"").append(entityFullId).append("\"")
+                .append("&rows=").append(maxResults)
+                .append("&sort=random")
+                .append("&profile=minimal")
+                .append("&wskey=").append(wskey);
+        return s.toString();
+    }
+
+    /**
+     * extracts items from set api response
+     * @param jsonObject
+     * @return
+     */
+    public List<String> extractItems(JSONObject jsonObject) {
+        List<String> items = new ArrayList<>();
+        try {
+            JSONArray setsList = jsonObject.getJSONArray(ITEMS_FIELD);
+            for(int i =0; i < setsList.length(); i++) {
+                JSONObject set = (JSONObject) setsList.get(i);
+                JSONArray itemList =  set.getJSONArray(ITEMS_FIELD);
+                for(int j =0; j < itemList.length(); j++) {
+                 items.add(StringUtils.substringAfter((String) itemList.get(j), "http://data.europeana.eu/item"));
+                }
+            }
+        } catch (JSONException e) {
+            LOG.error("Error fetching the objects from set api response. {}", e.getMessage());
+        }
+        return items;
+    }
+
+    /**
+     * Extracts skos:prefLabel(s) and skos:altLabel(s)
+     *
+     * @param jsonObject
+     * @return
+     */
+    public List<String> extractLabels(JSONObject jsonObject) {
+        List<String> labels = new ArrayList<>();
+        try {
+            JSONArray items =  jsonObject.getJSONArray(ITEMS_FIELD);
+            for(int i =0; i < items.length(); i++) {
+                JSONObject item = (JSONObject) items.get(i);
+                JSONObject prefLabel = item.getJSONObject(PREF_LABEL_FIELD);
+                JSONObject altLabel = item.getJSONObject(ALT_LABEL_FIELD);
+                labels.addAll(getPreflabels(prefLabel.toString()));
+                labels.addAll(getAltLabels(altLabel.toString()));
+            }
+        } catch (JSONException e) {
+            LOG.error("Error fetching the objects from entity api response. {}", e.getMessage());
```

Comment:
SQ issue: "Log or rethrow this exception"

Reply:
let the service class handle the exception